### PR TITLE
Add TorchVision wheel builds for M1

### PR DESCRIPTION
**Pull request by malfet**

### CODE REVIEW

**Comment by datumbox** [comment]
Also the [linter](https://app.circleci.com/pipelines/github/pytorch/vision/17813/workflows/3d7c8f18-c961-4d71-975e-dfeb5b238dd4/jobs/1442508) is complaining for a trailing new line that is missing.

File: .github/workflows/build-m1-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    paths:
+      - .github/workflows/build-m1-binaries.yml
+  push:
+    branches:
+      - nightly
+  workflow_dispatch:
+jobs:
+  build_wheels:
+    name: "Build TorchVision M1 wheels"
+    runs-on: macos-m1
```
Comment (atalman) [inline .github/workflows/build-m1-binaries.yml:13]: We have only 1 macos-m1 worker currently. I will change the rest of our workers to adhere to the same macos-m1 label

**Comment by NicolasHug** [comment]
@malfet @atalman 

as @datumbox noted above https://github.com/pytorch/vision/pull/5948#issuecomment-1134505355, we're not certain that torchvision properly works on M1. In order to avoid additional bug reports from users, are you planning to add an M1 CI so we're sure the binaries are working properly?

**Comment by malfet** [comment]
> In order to avoid additional bug reports from users, are you planning to add an M1 CI so we're sure the binaries are working properly?

@NicolasHug we don't have a capacity to run PR CI on M1 right now (there are 4 machines in total for the entire org), but if you can pick a few smoketests we can try running them.

**Comment by datumbox** [comment]
@malfet Our users are currently reporting that multiple parts of TorchVision are not working on M1: https://github.com/pytorch/vision/issues/6067, https://github.com/pytorch/vision/issues/5893, https://github.com/pytorch/vision/issues/5413

If that's true, it will take more than a couple of smoketests to debug, fix and ensure it works smoothly. If we currently don't have the resources for adding proper support for M1 and testing them across PyTorch, we should revert this PR and consider adding it back once we have the necessary capacity. Otherwise, after release we will have a huge volume of users, reporting that TorchVision is broken on M1 with maintainers unable to know what to do next and how to address this. We are happy to discuss a middle ground solution that allows us to run our tests in one configuration on M1. Please let me know.

**Comment by malfet** [comment]
@datumbox thank you for sharing. All those sound like a basic packaging issues to me, so I'm not sure how CI testing (where system is configured to have all above mentioned dependencies will help) And even if PR CI would be enabled, those machines would not have rerun with SSH capability.

But we should definitely test for the conditions mentioned in above during packaging.

**Comment by datumbox** [comment]
I had an offline discussion with @malfet concerning this. I'll summarize here what was discussed:

Our team doesn't have an M1 mac locally but even if we did, we still need to test each PR prior merging on at least one M1/Python configuration. Nikita agreed that since TorchVision is the first to be onboarded on native M1, we can do that now. Nevertheless because PyTorch has only 4 machines, it is not guaranteed that the job will be maintained. To avoid having the job removed, TorchVision should dedicate time to speed up CI execution for its tests.

I will therefore use the approach on this PR to add a unit-test job for a single python configuration. See #6111

**Pull request merged**